lisa/sut_orchestrator/baremetal/features.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+
+from typing import TYPE_CHECKING, Any
+
+from lisa import features, schema
+from lisa.feature import Feature
+
+if TYPE_CHECKING:
+    from .platform_ import BareMetalPlatform
+
+
+class StartStop(Feature):
+    def __getattr__(self, key: str) -> Any:
+        assert self._inner, "inner is not set"  # type: ignore
+        return getattr(self._inner, key)  # type: ignore
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
```

Comment:
Create a base class for all features, so the logic won't be repeated on each feature.

Reply:
You can add an inherit3ed method to get _feature_type.

```python
class ClusterFeature(Feature):
    def __getattr__(self, key: str) -> Any:
        assert self._inner, "inner is not set"  # type: ignore
        return getattr(self._inner, key)  # type: ignore

    def _initialize(self, *args: Any, **kwargs: Any) -> None:
        _feature_type = self._get_feature_type()
        self._inner = _feature_type(
            schema.FeatureSettings.create(_feature_type.name()),
            self._node,
            self._platform,
            *args,
            **kwargs,
        )

class StartStop(ClusterFeature):
    def _get_feature_type(self) -> Feature:
        platform: BareMetalPlatform = self._platform  # type: ignore
       return self.platform.get_start_stop()
```